### PR TITLE
TST: add test for df construction from dict with tuples

### LINKED ISSUE #16769: DataFrame.from_dict unexpectedly "flatten" tuples in the dictionary keys
#### Code Sample, a copy-pastable example if possible

```python
In [1]: import pandas

In [2]: pandas.__version__
Out[2]: '0.20.2'

In [3]: pandas.DataFrame.from_dict([{('a',): 1}, {('a',): 2}]).columns
Out[3]: Index(['a'], dtype='object')

In [4]: pandas.DataFrame.from_dict([{('a',): 1, ('b',): 2}]).columns
Out[4]: Index([('a',), ('b',)], dtype='object')

In [5]: pandas.DataFrame.from_dict([{('a', 'b'): 1}]).columns
Out[5]: Index([('a', 'b')], dtype='object')
```

#### Problem description

When (1) dictionaries with a single identical key is given to `pandas.DataFrame.from_dict` and (2) the key is a singleton tuple, then it returns a dataframe whose column is the content of the tuple, instead of the tuple itself.

Note that this problem does not happen when (1) is not the case (see `In [4]`) or (2) is not the case (see `In [5]`).  It makes the case (1) & (2) inconsistent with those other cases.

#### Expected Output

```python
In [3]: pandas.DataFrame.from_dict([{('a',): 1}, {('a',): 2}]).columns
Out[3]: Index([('a',)], dtype='object')
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.0.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.47-1-lts
machine: x86_64
processor:
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.2
pytest: 3.1.2
pip: 9.0.1
setuptools: 36.0.1
Cython: None
numpy: 1.13.0
scipy: None
xarray: None
IPython: 6.1.0
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

It is also reproduced with the current master branch:

<details>
INSTALLED VERSIONS
------------------
commit: 1265c27f4bbd06e1bb75f846139a164bdadd5b31
python: 3.6.0.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.47-1-lts
machine: x86_64
processor:
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.21.0.dev+179.g1265c27f4
pytest: None
pip: 9.0.1
setuptools: 36.0.1
Cython: 0.25.2
numpy: 1.13.0
scipy: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #16769
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -1424,6 +1424,20 @@ def test_from_dict_columns_parameter(self):
                 dict([("A", [1, 2]), ("B", [4, 5])]), columns=["one", "two"]
             )
 
+    @pytest.mark.parametrize(
+        "keys, values",
+        [([("a",), ("a",)], [1, 2]), ([("a",), ("b",)], [1, 2]), ([("a", "b")], [1])],
+    )
+    def test_constructor_from_dict_tuples(self, keys, values):
+        data_dict = {key: value for key, value in zip(keys, values)}
```
Comment (WillAyd) [inline pandas/tests/frame/test_constructors.py]: Can you just parametrize dictionaries directly instead of doing this?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_constructors.py (hunk 1 above)
Comment (ganevgv) [reply pandas/tests/frame/test_constructors.py]: Thanks @WillAyd - incorporated the change.

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -1424,6 +1424,25 @@ def test_from_dict_columns_parameter(self):
...
+            [OrderedDict([(("a",), 1), (("b",), 2)])],
+            [{("a", "b"): 1}],
+        ],
+    )
+    def test_constructor_from_dict_tuples(self, data_dict):
+        # GH 16769
+        df = DataFrame.from_dict(data_dict)
+
+        result = df.columns
+        all_columns = [key for sub_dict in data_dict for key in sub_dict.keys()]
```
Comment (WillAyd) [inline pandas/tests/frame/test_constructors.py]: You can just parametrize the expected column values alongside the data instead of doing this

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_constructors.py (hunk 2 above)
Comment (ganevgv) [reply pandas/tests/frame/test_constructors.py]: got you - done

**Review by WillAyd — APPROVED** [review]
lgtm @mroeschke care to take a look?

**Comment by mroeschke** [comment]
Thanks @ganevgv!

**Pull request merged**